### DISCUSSION TITLE
Database insert errors

### DISCUSSION THREAD

**Discussion by mjksohonet** [netdata/netdata#15271] [Q&A]
Hi,
my error.log is full of messages like this
`2023-06-28 14:21:00: netdata ERROR : LIBUV_WORKER : Failed to execute 'INSERT OR REPLACE INTO host_info (host_id, system_key, system_value, date_created) VALUES (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_CONTAINER_OS_DETECTION','/etc/os-release', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_HOST_OS_NAME',''XCP-ng'', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_HOST_OS_ID','xenenterprise', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_HOST_OS_ID_LIKE','centos rhel fedora', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_HOST_OS_VERSION','8.2.1', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_HOST_OS_VERSION_ID','8.2.1', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_HOST_OS_DETECTION','/etc/os-release', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_KERNEL_NAME','Linux', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_KERNEL_VERSION','4.19.0 1', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_ARCHITECTURE','x86_64', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_VIRTUALIZATION','none', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_VIRT_DETECTION','systemd-detect-virt', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_CONTAINER','none', unixepoch()), (u2h('48a13908-e910-11ed-9da8-0cc47ac3afeb'), 'NETDATA_SYSTEM_CONTAINER_DETECTION','systemd-detect-virt', unixepoch())', rc = 1 (near "XCP": syntax error) -- attempt 0
`
Does this mean that no data is being persisted to disk?

**Comment by MrZammler** [comment]
Hi @mjksohonet 

Yes, it breaks because the `XCP-ng` contains `'`s (i.e. `'XCP-ng'`). Likely we need to escape those or remove them...

**Comment by ilyam8** [comment]
@mjksohonet please open a [bug report](https://github.com/netdata/netdata/issues/new/choose).

**Reply by mjksohonet** [reply]
https://github.com/netdata/netdata/issues/15273 created.
I'm very new to netdata and I'm wondering if there is a way to keep lots of metrics in memory?
At the moment we're only seeing about an hour's worth of history.

**Reply by andrewm4894** [reply]
For sure you can alter default configs for much much longer storage.

But if you are only seeing an hour then maybe that might be related to the issue above perhaps. As out of the box, on typical set ups (eg not special in some way or with tens of thousands of metrics). So I'd expect once the above bug resolved you should see a lot more retention out of the box by default 

https://learn.netdata.cloud/docs/configuring/optimizing-metrics-database/change-how-long-netdata-stores-metrics
